### ISSUE TITLE
Support Error

### ISSUE THREAD

**Issue by confuser** [tabone/ipc-emitter#1]
Currently emitting Error's results in an empty string. Most likely due to JSON.stringify/parse.

E.g.

``` js
worker.emit('something-went-wrong', new Error('test'))
```

**Comment by tabone** [comment]
You're right, internally this lib uses [process.send()](https://nodejs.org/api/process.html#process_process_send_message_sendhandle_options_callback) to send payloads (info about emitted events) between a `master` and its `worker`. If you take a  look at the docs of [process.send()](https://nodejs.org/api/process.html#process_process_send_message_sendhandle_options_callback):

> Note: This function uses JSON.stringify() internally to serialize the message.*

I would suggest to do the following instead:

``` javascript
worker.emit('something-went-wrong', { message: 'test' })
```

**Comment by confuser** [comment]
Your suggestion involves losing stack traces and other useful information.

Ideally this module would handle the stringification/rebuilding of the error.

**Comment by tabone** [comment]
Can't you include the `stack` info using the same technique I suggested?

``` javascript
var err = new Error('woops')
worker.emit('something-went-wrong', { message: err.message, stack: err.stack })
```

**Comment by confuser** [comment]
For every error event sure, but seems convoluted for a very common case.
I'm happy to open a PR to support it.

**Comment by tabone** [comment]
Yeah, you're right and you've convinced me. Don't worry I'll implement it myself :) I have some internal changes which I would like to first implement. I'll keep you updated!

**Comment by tabone** [comment]
Just committed the Internal changes mentioned in the last comment https://github.com/tabone/ipc-emitter/commit/e891327c05bf5a6b8db799948cb08704e838c33a. 

I'll work on this feature in the coming days :)

Thanks again!

**Comment by tabone** [comment]
This feature is being implemented on [marshaller branch](https://github.com/tabone/ipc-emitter/tree/marshaller).

**Comment by confuser** [comment]
Amazing. Thank you!

**Comment by tabone** [comment]
Feature released with v0.1.5 :)

Thanks for your suggestion. I'll be closing this issue, but feel free to reopen it!
